core/admin.py

```diff
@@ -12,36 +12,46 @@
 from . import tasks
 
 
-def _text_preview(log_file: FileField):
+def _text_preview(target_file: FileField, mimetype, show_end):
     """
     Return the text of a file if it is short or the last portion of it if it is long.
 
-    :params: log_file A FileField to read text from.
+    params: target_file A FileField to read text from.
+            mimetype Log or Data's mimetype, score result don't have one
+            show_end Determine if showing the first/last portion for preview
     """
+    mimetype_check = mimetype.startswith('text/')
     # max file size for display, currently 10kb
     maxlen = 10000
-    if log_file:
-        with log_file.open('rb') as datafile:
+    if target_file:
+        with target_file.open('rb') as datafile:
             if len(datafile) > 0:
-                try:
+                if show_end:
+                    # read and only display from the end for log filess                    display_message = 'last'
```

Comment:
It looks like `display_message = 'last'` has been combined with the previous line.